test/resources/user_auth.go

```diff
@@ -0,0 +1,196 @@
+package resources
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"github.com/google/go-querystring/query"
+	"golang.org/x/net/html"
+	"golang.org/x/net/publicsuffix"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"strings"
+)
+
+const (
+	OpenshiftTokenName = "openshift-session-token"
+	DefaultIDP         = "testing-idp"
+)
+
+// User used to create url user query
+type User struct {
+	Username string `url:"username"`
+	Password string `url:"password"`
+}
+
+// LoginOptions used to create and parse url login options
+type LoginOptions struct {
+	Client string `url:"client_id"`
+	IDP    string `url:"idp"`
+}
+
+// CallbackOptions used to create and parse url callback options
+type CallbackOptions struct {
+	Response string `url:"response_type"`
+	Scope    string `url:"scope"`
+	State    string `url:"state"`
+}
+
+// GetUserOpenshiftToken this function expects users and IDP to be created via `./scripts/setup-sso-idp.sh`
+func GetUserOpenshiftToken(oauthUrl string, masterURL string, idp string, username string, password string) (string, error) {
+	// declare transport
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+
+	// declare new cookie jar om nom nom
+	jar, err := cookiejar.New(&cookiejar.Options{PublicSuffixList: publicsuffix.List})
+	if err != nil {
+		return "", fmt.Errorf("error occured creating a new cookie jar : %w", err)
+	}
+
+	// declare http client
+	client := &http.Client{
+		Transport: tr,
+		Jar:       jar,
+	}
```

Comment:
Since more than one test may need an HTTP client, maybe the main test can initialise this instead and be passed to the common tests by context? This would make it easier to reuse. wdyt @ciaranRoche?